infra/cifuzz/config_utils.py

```diff
@@ -0,0 +1,124 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module for getting the configuration CIFuzz needs to run."""
+
+import logging
+import enum
+import os
+import json
+
+
+def _get_project_repo_name():
+  return os.path.basename(os.getenv('GITHUB_REPOSITORY'))
+
+
+def _get_pr_ref(event):
+  if event == 'pull_request':
+    return os.getenv('GITHUB_REF')
+  return None
+
+
+def _get_sanitizer():
+  return os.getenv('SANITIZER', 'address').lower()
+
+
+def _get_project_name():
+  # TODO(metzman): Remove OSS-Fuzz reference.
+  return os.getenv('OSS_FUZZ_PROJECT_NAME')
+
+
+class Config:  # pylint: disable=too-few-public-methods,too-many-instance-attributes
+  """Object containing constant configuration for CIFuzz."""
+
+  class Platform(enum.Enum):
+    """Enum representing the different platforms CIFuzz runs on."""
+    EXTERNAL_GITHUB = 0  # Non-OSS-Fuzz on GitHub actions.
+    INTERNAL_GITHUB = 1  # OSS-Fuzz on GitHub actions.
+    INTERNAL_GENERIC_CI = 2  # OSS-Fuzz on any CI.
+
+  def _get_config_from_event_path(self, event):
+    event_path = os.getenv('GITHUB_EVENT_PATH')
+    if not event_path:
+      return
+    with open(event_path, encoding='utf-8') as file_handle:
+      event_data = json.load(file_handle)
+    if event == 'push':
+      self.base_commit = event_data['before']
+      logging.debug('base_commit: %s', self.base_commit)
+    else:
+      self.pr_ref = 'refs/pull/{0}/merge'.format(
+          event_data['pull_request']['number'])
+      logging.debug('pr_ref: %s', self.pr_ref)
+
+    self.git_url = event_data['repository']['git_url']
+
+  def __init__(self):
+    """Get the configuration from CIFuzz from the environment. These variables
+    are set by GitHub or the user."""
+    self.project_name = _get_project_name()
+    self.project_repo_name = _get_project_repo_name()
+    self.commit_sha = os.getenv('GITHUB_SHA')
+
+    event = os.getenv('GITHUB_EVENT_NAME')
+    self.is_github = bool(event)
+    logging.debug('Is github: %s.', self.is_github)
+
+    self.pr_ref = None
+    self.git_url = None
+    self.base_commit = None
+    self._get_config_from_event_path(event)
+
+    self.base_ref = os.getenv('GITHUB_BASE_REF')
+    self.workspace = os.getenv('GITHUB_WORKSPACE')
+    self.project_src_path = get_project_src_path(self.workspace)
+
+    self.sanitizer = _get_sanitizer()
+    self.build_integration_path = os.getenv('BUILD_INTEGRATION_PATH')
+    self.allowed_broken_targets_percentage = os.getenv(
+        'ALLOWED_BROKEN_TARGETS_PERCENTAGE')
+    # Check if failures should not be reported.
+    self.dry_run = _is_dry_run()
+
+  @property
+  def platform(self):
+    """Returns the platform CIFuzz is runnning on."""
+    if self.build_integration_path:
+      return self.Platform.EXTERNAL_GITHUB
+    if self.is_github:
+      return self.Platform.INTERNAL_GITHUB
+    return self.Platform.INTERNAL_GENERIC_CI
+
+
+def _is_dry_run():
```

Comment:
nit: move to top with the rest of the env var reading functions.

Reply:
Done.